packages/esm-tb-app/src/api/api.ts

```diff
@@ -0,0 +1,31 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+
+async function fetchReportData(reportId: string) {
+  try {
+    const response = await openmrsFetch(`ws/rest/v1/mamba/report?report_id=${reportId}`);
+    const data = await response.json();
+
+    if (data && data.results && data.results.length > 0) {
+      const record = data.results[0].record;
+
+      for (const item of record) {
+        return item.value ? parseInt(item.value, 10) : 0;
+      }
+    }
+
+    return 0;
+  } catch (error) {
+    console.error(`Error fetching data for report_id=${reportId}: `, error);
+    throw new Error(`Error fetching data for report_id=${reportId}: ${error}`);
+  }
+}
```

Comment:
Please move this to the commons lib so it can be reused and rename it to `fetchMambaReportData` or change it to a hook and use SWR to take advantage of caching.